src/util/uint.rs

```diff
@@ -404,7 +404,11 @@ macro_rules! construct_uint {
             }
         }
 
-        $crate::internal_macros::display_from_debug!($name);
+        impl core::fmt::Debug for $name {
+            fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+                core::fmt::LowerHex::fmt(&self, f)
```

Comment:
Once `0x` is conditioned on `alternate()` this would not print `0x` which could be confusing. `write!("{:#x}") works for now, ideally we would later support padding.